modyn/trainer_server/internal/dataset/online_dataset.py

```diff
@@ -149,13 +156,23 @@ def _unpack_data_tuple(self, data_tuple: Tuple) -> Tuple[int, bytes, int, Option
     def _yield_samples(self, key: int, sample: bytes, label: int, weight: Optional[float]) -> Tuple:
         assert self._uses_weights is not None
         # mypy complains here because _transform has unknown type, which is ok
+        tranformed_sample = self._transform(sample)  # type: ignore
+        if self._tokenizer is not None:
+            tranformed_sample = self._tokenizer(tranformed_sample)
+
         if self._uses_weights:
-            return key, self._transform(sample), label, weight  # type: ignore
-        return key, self._transform(sample), label  # type: ignore
+            return key, tranformed_sample, label, weight
+        return key, tranformed_sample, label
 
     def end_of_trigger_cleaning(self) -> None:
         self._key_source.end_of_trigger_cleaning()
 
+    def _instantiate_tokenizer(self, tokenizer: str) -> Callable:
+        tokenizer_module = dynamic_module_import("modyn.models.tokenizers")
+        if not hasattr(tokenizer_module, tokenizer):
+            raise ValueError("Requested tokenizer is not available.")
+        return getattr(tokenizer_module, tokenizer)()
```

Comment:
We can use the instantiateclass function we introduced, right?

Reply:
Yep, I tried to keep the structure similar on purpose

Reply:
ahh it is not merged in main yet. maybe just merge #277 first then, since we need to order the PRs anyways, and then update that

Reply:
leaving that open so we do not forget until the other pRs are merged

Reply:
Changed accordingly